platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/LearnerPushVirtualTreeView.java

```diff
@@ -338,4 +340,25 @@ public void setChild(final Long parent, final int childIndex, final Long child)
     public Long convertMerkleRootToViewType(final MerkleNode node) {
         throw new UnsupportedOperationException("Nested virtual maps not supported");
     }
+
+    private boolean isLeaf(final long path) {
+        return path >= reconnectState.getFirstLeafPath() && path <= reconnectState.getLastLeafPath();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void recordHashStats(
```

Comment:
What about other `LearnerTreeView` implementations? There are at least two more: `LearnerPushVirtualTreeView` (push model, virtual trees) and `LearnerPullVirtualTreeView` (pull model, virtual trees)

Reply:
The current implementation is targeting VirtualMaps specifically because regular merkle trees are too small currently to make the metrics any useful. For this reason, the `LearnerPushMerkleTreeView` is omitted.

As for the pull model, it's currently disabled under a feature flag. Additionally, I don't have sufficient expertise in this implementation. I'd suggest you to update the pull-model implementation as appropriate in a separate PR.

Reply:
In the latest commit, I in fact implemented the method in `LearnerPushMerkleTreeView` in order to cover all the branches that we currently use in production.

Reply:
The pull classes aren't used by default in `develop`, but they are fully functional and should be treated as fully supported. It would be surprising for anybody to enable a pull model and find that it doesn't support new metrics... If it's tracked in a separate ticket, it should be worked immediately after this one, not some time later

Reply:
Okay, apparently it had an implementation for the `isLeaf` already, so I overrode the method in the `LearnerPullVirtualTreeView` as well.